### ISSUE TITLE
Laggy mouse control in Windows version

### ISSUE THREAD

**Issue by instr3** [SerGreen/TheWitnessPuzzles#13]
Hi,

We found that the mouse control is a bit laggy after generating and playing several puzzles (do not need to solve them, we just tried them and skipped them). The behavior is getting worse after a few puzzles (maybe 5-10 puzzles): The line grows at a very slow, unsmoothed speed even if I drag my mouse very fast. 

I also sent it to my friend and they confirm that this also happens to them. I looked into the code but I did not find any clue why this happened. 

I wonder if others can reproduce this issue and/or look into this problem?

Thanks!

**Comment by SerGreen** [comment]
Weird, i don't notice any difference on my side.  

https://user-images.githubusercontent.com/4000373/103446511-0ef37980-4c89-11eb-90f5-b39c953dd0e2.mp4

What settings did you use? Bloom FX, fullscreen, mouse sensitivity? Does it go back to normal after you close the game and launch again? Does it get progressively worse every few puzzles or does it suddenly become slow once and doesn't get even slower after that?  

For now, unfortunately, i have no idea what can cause such slowdown. It would be great to see this bug in action. If it's possible for you to record a clip, i would appreciate that (for example, Windows 10 has a built-in recorder in `Win+G` Game Bar).

**Comment by instr3** [comment]
Hi SerGreen,

Thank you for your fast response! I just found that one fast way to reproduce it is to resize the window. I do not know if resizing the window is the cause for it, but it might be a clue.

To reproduce the issue:
1. Start the game, and we can draw lines smoothly
2. Cancel the line drawing and resize the window
3. Try to draw lines again and it might become laggy. If it does not become laggy, try step 2 again.
4. If we restart the program (or sometimes just wait for some seconds), things become smooth again.

I did not turn on BloomFX nor the fullscreen mode. I also did not change the mouse sensitivity. 

I recorded two clips using ``Win+G`` game bar's recorder (since resizing the window will cause the recorder to stop), the first one is the case where we can smoothly draw lines, and the second one is the laggy situation (and it seems to recover after some seconds). My computer's CPU, GPU, and memory usage are low in both situations.



https://user-images.githubusercontent.com/13694510/103452604-17d95f00-4d0c-11eb-998f-5a896a1c492c.mp4

https://user-images.githubusercontent.com/13694510/103452606-1c057c80-4d0c-11eb-8233-738dfb4df772.mp4

**Comment by SerGreen** [comment]
Oh yes, resizing absolutely screws it up for me too. Thanks for the report, i'll look into it.  
For now as a workaround you can exit to main menu and back to puzzle to fix the lag, it seems.

**Comment by SerGreen** [comment]
TL;DR: Please try [v1.0.8.1 test release](https://github.com/SerGreen/TheWitnessPuzzles/releases/tag/v1.0.8.1) and tell me if it's fixed for you.  

This issue is weird.  
I tried it on an old weak laptop and i couldn't manage to make mouse lag there at all. On my much stronger PC it lags 9 times out of 10 for a few seconds after window resize and then goes back to normal (but rarely it stays laggy indefinitely). Afterburner shows 60 fps but uneven frame times during these lags.  

It seems to be hardware/monogame related. [This forum thread](https://community.monogame.net/t/solved-jittery-choppy-frame-movement-for-2d-game/11888) mentions similar issues, so i just tried to slap on `IsFixedTimeStep` and `SynchronizeWithVerticalRetrace` and see what happens.  
With VSync off it's just laggy all the time, but with added FixedTimeStep = false it went up to 3000+ fps and became super smooth. But 3k fps is kind of a waste of resources, so lastly i tried VSync on and FixedTimeStep off and i wasn't able to make it lag again after that, so i suppose this might be the fix, but i can't be sure.  
So i published a [preview-release of windows version](https://github.com/SerGreen/TheWitnessPuzzles/releases/tag/v1.0.8.1). Can you please test it on your machine and report if the bug is still there?  

Meanwhile i want to (finally) add an ability to enter seed for a panel, so it'd be possible to share interesting panels at least by typing in a seed number. Seeds were there for a long time, but there's no way in the UI to provide a seed for panel generator.  
When it's ready, i'll publish it with a fix for this bug included as a proper new release.

**Comment by instr3** [comment]
The preview-release seems to work fine for me! It does not lag even after tens of tries.

Also, glad to know that there will be new features in the next version!
